### PR TITLE
 Make all members of HTTP{Request|Response}Head mutable.

### PR DESCRIPTION
**Pull request by tanner0101**
Make all members of HTTP{Request|Response}Head mutable.

### Motivation:

Immutable members of structs (properties defined with let) can never be changed
even if the struct itself is mutable (defined withh var). This makes it difficult
to build rich convenience APIs on top of these structs.

### Modifications:
Changed all 'let' stored properties on HTTP{Request|Response}Head to 'var'.
### Result:

HTTP{Request|Response}Head are easier to use now.

### CODE REVIEW

**Comment by Lukasa** [comment]
@swift-nio-bot add to whitelist

File: Sources/NIOHTTP1/HTTPTypes.swift [hunk 1]
```diff
@@ -20,18 +20,18 @@ let headerSeparator: StaticString = ": "
 /// A representation of the request line and header fields of a HTTP request.
 public struct HTTPRequestHead: Equatable {
     /// The HTTP method for this request.
-    public let method: HTTPMethod
+    public var method: HTTPMethod
 
     // Internal representation of the URI.
-    private let rawURI: URI
+    private var rawURI: URI
```
Comment (normanmaurer) [inline Sources/NIOHTTP1/HTTPTypes.swift]: This one is private so I think this makes no sense or I am missing something ?

Comment (Lukasa) [reply Sources/NIOHTTP1/HTTPTypes.swift]: Yeah, that one could stay `let`.

Comment (tanner0101) [reply Sources/NIOHTTP1/HTTPTypes.swift]: I missed adding a `set` case to `uri`, that is added now (and relies on rawURI being mutable)

File: Sources/NIOHTTP1/HTTPTypes.swift [hunk 2]
```diff
@@ -20,18 +20,19 @@ let headerSeparator: StaticString = ": "
...
 
     /// The URI used on this request.
     public var uri: String {
-        return String(uri: rawURI)
+        get { return String(uri: rawURI) }
```
Comment (weissi) [inline Sources/NIOHTTP1/HTTPTypes.swift]: we don't usually skip the newlines here but I don't mind. CC @normanmaurer / @Lukasa

**Comment by weissi** [comment]
@swift-nio-bot test this please

File: Sources/NIOHTTP1/HTTPTypes.swift (hunk 2 above)
Comment (normanmaurer) [reply Sources/NIOHTTP1/HTTPTypes.swift]: I would prefer to add them just to make things consistent .

Comment (tanner0101) [reply Sources/NIOHTTP1/HTTPTypes.swift]: Consistency is king

**Comment by normanmaurer** [comment]
@tanner0101 thanks!

**Pull request merged**